### PR TITLE
Add revision secondary sort to `resolve_latest_keg`

### PR DESCRIPTION
**Pull request by cnnrmnn**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Context for this change is [here](https://github.com/Homebrew/brew/pull/11824/files#r684843341).

### CODE REVIEW

File: Library/Homebrew/cli/named_args.rb [hunk 1]
```diff
@@ -327,7 +327,11 @@ def resolve_latest_keg(name)
 
         stable_kegs = kegs.reject { |k| k.version.head? }
 
-        return kegs.max_by { |keg| Tab.for_keg(keg).source_modified_time } if stable_kegs.blank?
+        if stable_kegs.blank?
+          return kegs.max_by do |keg|
+                   [Tab.for_keg(keg).source_modified_time, keg.version.revision]
+                 end
```
Comment (Rylan12) [inline Library/Homebrew/cli/named_args.rb:333]:
```suggestion
          return kegs.max_by do |keg|
            [Tab.for_keg(keg).source_modified_time, keg.version.revision]
          end
```

Just a personal preference thing but I'd say this is more consistent with the rest of the codebase. Not a functional difference though so I'd say up to whatever you prefer.

**Review by Rylan12 — APPROVED** [review]
Thanks, @cnnrmnn!

File: Library/Homebrew/cli/named_args.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/cli/named_args.rb:333]: Agreed 👍🏻

Comment (cnnrmnn) [reply Library/Homebrew/cli/named_args.rb:333]: Fine by me. I ran `brew style --fix` on slightly different formatting and it produced the above formatting.

**Pull request merged**